### PR TITLE
fix CI error

### PR DESCRIPTION
**Pull request by asherascout**
Hello wim!

I made some changes to fix CI error.

- Delete linters that disabled on go1.18 temporally. It's not necessary actually because we still use go1.15 in CI. But I think removing golint is ok because it's deprecated and frozen.
- Upgrade golangci-lint version.
- Fix golanci-lint warning.
- Refactor two functions which is longer than 80. I'll change it after your review if you think it is not good.

### CODE REVIEW

**Comment by asherascout** [comment]
I don't know how to resolve conflicts. 🌝

File: .golangci.yml [hunk 1]
```diff
@@ -115,22 +112,19 @@ linters:
     - gofmt
     - gofumpt
     - goheader
-    - golint
```
Comment (wimspaargaren) [inline .golangci.yml:118]: Can we replace this one with `revive` ?

**Review by wimspaargaren — APPROVED** [review]
Overall nice improvement 🚀  Thx for the genuine interest in the tool!

File: .golangci.yml (hunk 1 above)
Comment (asherascout) [reply .golangci.yml:118]: I add revive and put golint in the disable list.

**Pull request merged**